packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -0,0 +1,352 @@
+  const _MountainViewPageTransition({
+    Key key,
+    this.animation,
+    this.secondaryAnimation,
+    this.child,
+  }) : super(key: key);
+
+  // The new page slides upwards just a little as its clip
+  // rectangle exposes the page from bottom to top.
+  static final Tween<Offset> _primaryTranslationTween = Tween<Offset>(
+    begin: const Offset(0.0, 0.05),
+    end: Offset.zero,
+  );
+
+  // The old page slides downwards a little as the new page appears.
+  static final Tween<Offset> _secondaryTranslationTween = Tween<Offset>(
+    begin: Offset.zero,
+    end: const Offset(0.0, 0.025),
+  );
+
+  // The scrim obscures the old page by becoming increasingly opaque.
+  static final Tween<double> _scrimOpacityTween = Tween<double>(
+    begin: 0.0,
+    end: 0.25,
+  );
+
+  // Used by all of the transition animations.
+  static const Curve _transitionCurve = Cubic(0.20, 0.00, 0.00, 1.00);
+
+  final Animation<double> animation;
+  final Animation<double> secondaryAnimation;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (BuildContext context, BoxConstraints constraints) {
+        final Size size = constraints.biggest;
+
+        // Gradually expose the new page from bottom to top.
+        final Animation<double> clipAnimation = Tween<double>(
+          begin: 0.0,
+          end: size.height,
+        ).animate(
+          CurvedAnimation(
+            parent: animation,
+            curve: _transitionCurve,
+            reverseCurve: _transitionCurve.flipped,
+          ),
+        );
+
+        final Animation<double> opacityAnimation = _scrimOpacityTween.animate(
+          CurvedAnimation(
+            parent: animation,
+            curve: _transitionCurve,
+            reverseCurve: _transitionCurve.flipped,
+          ),
+        );
+
+        final Animation<Offset> primaryTranslationAnimation = _primaryTranslationTween.animate(
+          CurvedAnimation(
+            parent: animation,
+            curve: _transitionCurve,
+            reverseCurve: _transitionCurve.flipped,
+          ),
+        );
+
+        final Animation<Offset> secondaryTranslationAnimation = _secondaryTranslationTween.animate(
+          CurvedAnimation(
+            parent: secondaryAnimation,
+            curve: _transitionCurve,
+            reverseCurve: _transitionCurve.flipped,
+          ),
+        );
+
+        return AnimatedBuilder(
+          animation: animation,
+          builder: (BuildContext context, Widget _) {
+            return Container(
+              color: Colors.black.withOpacity(opacityAnimation.value),
+              alignment: Alignment.bottomLeft,
+              child: ClipRect(
+                child: SizedBox(
+                  height: clipAnimation.value,
+                  child: OverflowBox(
+                    alignment: Alignment.bottomLeft,
+                    maxHeight: size.height,
+                    child: AnimatedBuilder(
```

Comment:
you can make this a `child` of the outer AnimatedBuilder to avoid rebuilding the inner one each time

Reply:
Good point! Done.